provisioner/unpacker/main.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+type unpackOptions struct {
+	BundleName       string
+	StorageNamespace string
```

Comment:
Maybe this is premature optimization but I think we could add `StorageType` or something here that would represent that this unpacker unpacks to a ConfigMap specifically. Or simply call it ConfigMapUnpacker.

Reply:
Agreed, I was thinking the same thing and I've been playing around something locally that attempts to fire off Jobs based on the type of Bundle source being reconciled. I landed on avoiding trying to make this Job look like fully fleshed out helm chart template, but I think that's a reasonable path forward long-term, once more Bundle sources are supported.